server/api.js

```diff
@@ -69,20 +64,74 @@ router.delete("/pages/:page_title/:record_id", async (req, res) => {
 		const record_id = req.params.record_id;
 
 		const findPageId = await db.query(
-			`select page_id from pages where page_title = $1`,[pageTitle]
+			"select page_id from pages where page_title = $1",
+			[pageTitle]
 		);
 		const page_id = findPageId.rows[0].page_id;
 		const deletingModule = await db.query(
 			"delete from modules WHERE page_id = $1 AND record_id = $2",
 			[+page_id, +record_id]
 		);
 		res.status(200).json(deletingModule[0].rows);
-
 	} catch (err) {
-		console.error(err);
-	res.status(500).json({ error: err.message });
+		res.status(500).json({ error: err.message });
+	}
+});
+
+// add new page endpoint
+
+router.post("/pages/:newpage", async (req, res) => {
+	const newpage = req.params.pageTitle;
+	const { page_id, page_title, page_path } = req.body;
+
+	if (!page_id || !page_title || !page_path) {
+		res.status(500).json("Please fill all the fields");
+	}
+
+	try {
+		// introduce a new row into the pages table
+		const nwePage = await db.query(
+			`INSERT INTO pages (page_id,page_title, page_path)
+      VALUES ($1, $2, $3)`,
+			[page_id, page_title, page_path]
+		);
+
+		// Second inserting the above data into the pages table
+		const insertIntoPagesTable = await db.query(
```

Comment:
with this you will endup with two pages for each add new button. because you are inserting in line 94 and line 101

Reply:
if I understand it correct this code will generate a row the add data into  `// introduce a new row into the pages table
		const nwePage = await db.query(
			`INSERT INTO pages (page_title, page_path)
      VALUES ($1, $2,)`,
			[page_title, page_path]
		);`

Reply:
yes correct. so if you run it twice, it will add the row twice.